src/OrbitGl/include/OrbitGl/CallTreeView.h

```diff
@@ -188,20 +188,42 @@ class CallTreeUnwindErrorType : public CallTreeNode {
   orbit_client_data::CallstackType error_type_;
 };
 
-class CallTreeView : public CallTreeNode {
+class CallTreeView {
  public:
   [[nodiscard]] static std::unique_ptr<CallTreeView> CreateTopDownViewFromPostProcessedSamplingData(
       const orbit_client_data::PostProcessedSamplingData& post_processed_sampling_data,
-      const orbit_client_data::ModuleManager& module_manager,
-      const orbit_client_data::CaptureData& capture_data);
+      const orbit_client_data::ModuleManager* module_manager,
+      const orbit_client_data::CaptureData* capture_data);
 
   [[nodiscard]] static std::unique_ptr<CallTreeView>
   CreateBottomUpViewFromPostProcessedSamplingData(
       const orbit_client_data::PostProcessedSamplingData& post_processed_sampling_data,
-      const orbit_client_data::ModuleManager& module_manager,
-      const orbit_client_data::CaptureData& capture_data);
+      const orbit_client_data::ModuleManager* module_manager,
+      const orbit_client_data::CaptureData* capture_data);
+
+  CallTreeView(const orbit_client_data::ModuleManager* module_manager,
+               const orbit_client_data::CaptureData* capture_data)
+      : module_manager_{module_manager}, capture_data_{capture_data} {}
 
-  CallTreeView() : CallTreeNode{nullptr} {}
+  [[nodiscard]] CallTreeNode* GetCallTreeRootNode() { return &call_tree_root_; }
+  [[nodiscard]] const CallTreeNode* GetCallTreeRootNode() const { return &call_tree_root_; }
+
+  [[nodiscard]] const orbit_client_data::ModuleManager& GetModuleManager() const {
+    ORBIT_CHECK(module_manager_ != nullptr);
+    return *module_manager_;
+  }
+
+  [[nodiscard]] const orbit_client_data::CaptureData& GetCaptureData() const {
+    ORBIT_CHECK(capture_data_ != nullptr);
+    return *capture_data_;
+  }
+
+  [[nodiscard]] uint64_t sample_count() const { return call_tree_root_.sample_count(); }
+
+ private:
+  CallTreeNode call_tree_root_{nullptr};
```

Comment:
Oooh, now I remember what the idea was. `CallTreeView` as a subclass of `CallTreeNode` represented _the_ root of a call tree. Now that `CallTreeView` is something different, you are using the base class `CallTreeNode` for that. This is very confusing to me now, as I was wondering why this is not a pointer despite it being a base class and hence possibly having to hold instances of subclasses. You should add a subclass of `CallTreeNode` called `CallTreeRoot`, and make `CallTreeNode` abstract (just add a pure virtual destructor).

Reply:
Done.